### PR TITLE
Introduces ApplicationContextRunner for auto-configuration tests

### PR DESCRIPTION
**Pull request by bijukunjummen**
This is an attempt at using ApplicationContextRunner for some of the auto-configuration tests as a replacement for the more elaborate `@SpringBootTest` and could serve as a model for future tests. It feels faster and cleaner than the `@SpringBootTest`

### CODE REVIEW

**Comment by spencergibb** [comment]
Hmm, there are places where we don't use spring boot test and are using spring application.run, seems like those would be better places for enhancement

**Comment by bijukunjummen** [comment]
Possibly @spencergibb, I was debating what a good fit for such a test could be and picked some of the auto-configuration tests at random just to see how it fits in. Will scan around a little more and see where else it may fit in better.

File: spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/ribbon/RibbonAutoConfigurationIntegrationTests.java [hunk 1]
```diff
@@ -17,46 +17,40 @@
 
 package org.springframework.cloud.netflix.ribbon;
 
+import com.netflix.client.config.CommonClientConfigKey;
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/ribbon/RibbonAutoConfigurationIntegrationTests.java]: Import order

File: spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/ribbon/RibbonClientPreprocessorIntegrationTests.java [hunk 2]
```diff
@@ -17,65 +17,77 @@
...
-import org.springframework.context.annotation.Import;
-import org.springframework.test.annotation.DirtiesContext;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
-
 import com.netflix.loadbalancer.ConfigurationBasedServerList;
 import com.netflix.loadbalancer.DummyPing;
 import com.netflix.loadbalancer.Server;
 import com.netflix.loadbalancer.ZoneAvoidanceRule;
 import com.netflix.loadbalancer.ZoneAwareLoadBalancer;
+import org.junit.Test;
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/ribbon/RibbonClientPreprocessorIntegrationTests.java]: import order

File: spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/ribbon/RibbonDisabledTests.java [hunk 3]
```diff
@@ -19,21 +19,28 @@
...
-	@Test(expected = ArrayStoreException.class)
+	@Test
 	public void testRibbonDisabled() {
-		new SpringApplicationBuilder().web(false).sources(RibbonAutoConfiguration.class)
-				.run().close();
+		assertThatThrownBy(() -> new ApplicationContextRunner()
+				.withConfiguration(AutoConfigurations.of(RibbonAutoConfiguration.class))
+				.run(context -> assertThat(
+						context.getBeanNamesForType(SpringClientFactory.class))
+								.hasSize(0))).hasCauseExactlyInstanceOf(
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/ribbon/RibbonDisabledTests.java]: formatting is weird here

**Comment by bijukunjummen** [comment]
Yes @spencergibb , something in my local intellij configuration is messing up import order, will fix it and send an updated PR.

**Comment by bijukunjummen** [comment]
Sorry, closing this, my changes are badly conflicting with the changes to the master. Will send another PR later.

**Pull request closed without merging**